src/core/tsi/s2a/s2a_constants.h

```diff
@@ -61,6 +61,9 @@ constexpr char kS2AApplicationProtocol[] = "grpc";
  * **/
 constexpr size_t kTlsSequenceSize = 8;
 
+/** The maximum size of a frame expected by the S2A frame protector. **/
+constexpr size_t kS2AMaxFrameSize = 16 * 1024 * 1024;
```

Comment:
You have SSL3_RT_MAX_PLAIN_LENGTH, which is 16KB, but here you have max frame size to be 16MB?

Reply:
Good catch thanks! It should be: 

`/*record_header=*/5 + /*max_plaintext_size=*/16*1024 + /*tag_size=*/16`.

I've updated the constants file accordingly.